### PR TITLE
sentinel1

### PR DESCRIPTION
**Pull request by bhbraswell**
It's a mess, needs cleanup, but seems to work. Downloads are cached to make subsequent processing quicker. There's a lot of hard-coded stuff. tiles.py could be about half the length. Needs cleanup of the /stage directory.

gips_export sentinel1 -p sigma0 -d 2018-04-15 -s gips/test/NHseacoast.shp -v4 --fetch --res 20 20 --notld --outdir /archive/export/test_NH

Don't worry about merging but have a look and feel free to make suggestions.

### CODE REVIEW

**Comment by bhbraswell** [comment]
@subitc, the following test basically fails

`gips_process sentinel1 -p sigma0 -d 2018-100,2018-200 -t 531-390 -v4 --fetch`

It made it through 12 dates successfully, but then the process stopped. Container was frozen and I couldn't exec into it. Also I couldn't stop the container or remove it. Had to restart docker. Tough to say exactly what the problem was but probably not disk since I'm mounting all the storage locally.

**Comment by subitc** [comment]
> @subitc, the following test basically fails
> 
> `gips_process sentinel1 -p sigma0 -d 2018-100,2018-200 -t 531-390 -v4 --fetch`
> 
> It made it through 12 dates successfully, but then the process stopped. Container was frozen and I couldn't exec into it. Also I couldn't stop the container or remove it. Had to restart docker. Tough to say exactly what the problem was but probably not disk since I'm mounting all the storage locally.

I'm running it on S3. Will report back on what I find.

**Comment by bhbraswell** [comment]
I had to make a couple changes. You probably noticed that rtree needs to installed. Also the make_rectangular_tilegrid function was being called incorrectly.

**Review by subitc — APPROVED** [review]
Works for me now with the changes. We can make it leaner but no issue so big that we can't merge it now.

**Comment by bhbraswell** [comment]
Taking so much more than 8 hours to run on my laptop, but it keeps going. Before merging I just want to add a switch to the container build so that you don't have to install snap in order to get gips.

**Comment by bhbraswell** [comment]
Actually, this is a problem
```
rob@MBP-18-111:/mnt/storage/gips/sentinel1/tiles/531-390$ diff ./2018188/S1_IW_GRDH_20180707_531-390.tif ./2018187/S1_IW_GRDH_20180706_531-390.tif
```
Most of the dates look good and are different from one another, but I noticed this one, subsequent days with literally identical values.

**Comment by bhbraswell** [comment]
@sullamenashe since gips main is production I think someone on engineering should do a final review and merge. This optionally adds a sentinel1 gips driver to the container. To include it, you have to build with `--build-arg SENTINEL1=YES`. It adds a lot of stuff. What I think that means is that by default it shouldn't affect you until we want to include S1 in octopus, e.g. for flooding or whatever.

**Comment by bhbraswell** [comment]
OK I think this is ready. Removed the "[WIP]".

File: gips/data/sentinel1/sentinel1.py
Comment (sullamenashe) [inline gips/data/sentinel1/sentinel1.py:191]: What are the rate limits to downloading using their API?  Is it something ridiculous like 10 downloads at at time?

File: gips/data/sentinel1/sentinel1.py
Comment (sullamenashe) [inline gips/data/sentinel1/sentinel1.py:388]: What's the output tile grid look like?  Is it the same as Sentinel2?

Comment (bhbraswell) [reply gips/data/sentinel1/sentinel1.py:388]: Since S1 doesn't have a tile system I'm using our analysis tile boundaries, but at 20 meter resolution.

File: gips/data/sentinel1/sentinel1.py
Comment (bhbraswell) [reply gips/data/sentinel1/sentinel1.py:191]: Good question. All I know is that it seems slow.

**Review by sullamenashe — APPROVED** [review]

**Pull request merged**
